### ISSUE TITLE
Deprecated GoogleMapController.setMapStyle prevents toggling theme modes

### ISSUE THREAD

**Issue by bzark** [flutter/flutter#144524]
In versions pior to 2.6.0 I was able to toggle the dark/light themes using the following code:

```
await mapController.setMapStyle(isDarkMode ? await rootBundle.loadString('assets/google_maps_dark_mode.json') : '[]');
```

With the new update the `style` is set on the GoogleMap widget constructor.  When `setMapStyle` is finally removed from a future release how can one update their apps dark/light theme without restarting the app?

**Comment by timbotimbo** [comment]
In the PR (https://github.com/flutter/packages/pull/6192 )  they mention
>  since presumably people are defining their styles at build time, not runtime,

> Since we don't really want two recommended ways of doing the same thing, this deprecates the existing setMapStyle method in favor of the new approach.

Changing the style at runtime to handle changes in the app theme, does sound like a different use case than simply initializing the map.
The `setMapStyle` function ensures that you don't change any map state like zoom/location/rotation/etc.  
With the new approach you would have to rebuild the map to change the style.

**Comment by huycozy** [comment]
Hi @bzark 

>how can one update their apps dark/light theme without restarting the app?

The app will not be restarted unless you put GoogleMap widget to the root widget on your app, it depends on your implementation with re-setting map style when the theme changes. Or is this a typo? Do you mean `GoogleMap widget` in stead of `app`?

**Comment by bzark** [comment]
> Hi @bzark
> 
> > how can one update their apps dark/light theme without restarting the app?
> 
> The app will not be restarted unless you put GoogleMap widget to the root widget on your app, it depends on your implementation with re-setting map style when the theme changes. Or is this a typo? Do you mean `GoogleMap widget` in stead of `app`?

Yes, I meant how do we change the `GoogleMap` widgets theme without restarting the app.  Like I previously mentioned the `setMapStyle` was a perfect solution for this without needing to restart your app.

**Comment by timbotimbo** [comment]
I just tested the latest google maps version and it looks like the new parameter can handle the same desired behavior.

Taking this pseudo-code:

```
String _mapStyle = "..."

build() {
    ...
     GoogleMap(
        style:  _mapStyle,
    )
}
```

If I call setState and change the _mapStyle variable to a different style, the map transitions to the new style while keeping all state like location, zoom and rotation.

**Comment by bzark** [comment]
> I just tested the latest google maps version and it looks like the new parameter can handle the same desired behavior.
> 
> Taking this pseudo-code:
> 
> ```
> String _mapStyle = "..."
> 
> build() {
>     ...
>      GoogleMap(
>         style:  _mapStyle,
>     )
> }
> ```
> 
> If I call setState and change the _mapStyle variable to a different style, the map transitions to the new style while keeping all state like location, zoom and rotation.

Ah you are correct.  Thank you very much.  I'm not sure why I didn't think of this being I'm doing something similar with `MapType` when toggling normal and satellite types.
